File: UrbanDictionary/app/src/main/java/com/nike/urbandictionary/networking/UrbanDictionaryRetrofitService.java
package com.nike.urbandictionary.networking;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

class UrbanDictionaryRetrofitService {

    private static Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(" https://mashape-community-urban-dictionary.p.rapidapi.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    public static <S> S createService(Class<S> serviceClass) {
        return retrofit.create(serviceClass);
    }
}

File: UrbanDictionaryKotlin/app/src/main/java/com/nike/urbandictionary/networking/UrbanDictionaryRetrofitService.kt
package com.nike.urbandictionary.networking

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

internal object UrbanDictionaryRetrofitService {

    private val retrofit = Retrofit.Builder()
            .baseUrl(" https://mashape-community-urban-dictionary.p.rapidapi.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

    fun <S> createService(serviceClass: Class<S>): S {
        return retrofit.create(serviceClass)
    }
}
